functions/internal/util/util.go

```diff
@@ -176,3 +181,53 @@ func ReadCryptoRandBytes(b []byte) {
 		panic(fmt.Errorf("could not read random bytes: %v", err))
 	}
 }
+
+// CheckHTTPS retrieves the scheme from the X-Forwarded-Proto or RFC7239
+/*
+We do this because in the function running in a cloud container the SSL termination
+has likely happened upstream so we need to check the headers to reject HTTP only.
+Requests on GCE contain both of these headers and anything supplied by the client is
+overwritten. Locally in development mode we don't use HTTPS so the client should send
+one of these headers.
+*/
+
+var (
+	// De-facto standard header keys.
+	xForwardedProto = http.CanonicalHeaderKey("X-Forwarded-Proto")
+	forwarded       = http.CanonicalHeaderKey("Forwarded") // RFC7239
+	
+	protoRegex 		= regexp.MustCompile(`(?i)(?:proto=)(https|http)`)
+)
+
+func CheckHTTPS(r *http.Request) StatusError {
+	var scheme string
+
+	// Retrieve the scheme from X-Forwarded-Proto.
+	if proto := r.Header.Get(xForwardedProto); proto != "" {
+		scheme = strings.ToLower(proto)
+	} else if proto = r.Header.Get(forwarded); proto != "" {
+		// match should contain at least two elements if the protocol was
+		// specified in the Forwarded header. The first element will always be
+		// the 'proto=' capture, which we ignore. In the case of multiple proto
+		// parameters (invalid) we only extract the first.
+		if match := protoRegex.FindStringSubmatch(proto); len(match) > 1 {
```

Comment:
To be extra safe, why don't we check `len(match) == 2` and if `len(match) > 2`, we return a 500 and log an error? That way, if it turns out that our assumption that there can't be multiple `proto=` parameters is wrong, we catch it and can fix our code.